### PR TITLE
Add release notes for NuttX-10.0.0

### PR DESCRIPTION
**Pull request by btashton**
## Summary
This contains the release notes for the 10.0.0 release.

### CODE REVIEW

**Comment by btashton** [comment]
This is currently fully in sync with the draft in the wiki which is also up-to-date as of now.  There is one outstanding ticket that I know of that needs to be addressed before we merge this in and cut an RC.

**Comment by Ouss4** [comment]
Did you manually convert it to markdown?

We need to do the same to the [template](https://cwiki.apache.org/confluence/display/NUTTX/Release+Notes+Template+-+NuttX+X.Y) and then be able to just copy paste from Confluence.

**Comment by btashton** [comment]
> Did you manually convert it to markdown?
> 
> We need to do the same to the [template](https://cwiki.apache.org/confluence/display/NUTTX/Release+Notes+Template+-+NuttX+X.Y) and then be able to just copy paste from Confluence.

I agree it was annoying and I would say I did a marginal/okay job.  Basically I exported the confluence page as HTML then used turndown https://domchristie.github.io/turndown/ tweaked it a bit and ran it through rewrap to get the line lengths under control.

It would be better if we just did it in markdown in the first place.  I think the issues there might be I dont know how well Confluence lets multiple people edit at the same place.  There might be a better solution for that.  Something we can think about going forward.

**Comment by Ouss4** [comment]
> It would be better if we just did it in markdown in the first place. I think the issues there might be I dont know how well Confluence lets multiple people edit at markdown the same time. There might be a better solution for that. Something we can think about going forward.

Are you aware of any limitations in Confluence about that?  For a reference, podling reports are all in Markdown.  When submitted, it's just one file that's being edited by multiple people at the same time.
Anyway, we'll talk about this (and hopefully the ReleaseNote file too) after we get 10.0 out.

File: ReleaseNotes [hunk 1]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1597](https://github.com/apache/incubator-nuttx/pull/1597) sched: Support
+ passing non empty argument to init task
+
+ * [PR-1596](https://github.com/apache/incubator-nuttx/pull/1596) libc: Replace
+ all sem\_xxx with \_SEM\_XXX. This insures the correct semaphore interface is
+ used by userspace and the kernel.
+
+ * [PR-1517](https://github.com/apache/incubator-nuttx/pull/1517) sched/wdog:
+ Chang the default value of MAX\_WDOGPARMS from 4 to 2 as wd\_start is two every
```
Comment (acassis) [inline ReleaseNotes:27238]: Chang = Change

File: ReleaseNotes [hunk 2]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ building external code into the OS
+
+ * [PR-1396](https://github.com/apache/incubator-nuttx/pull/1396) Make C/C++
+ search path common so all boards support uClibc++/libc++ automatically
+
+ * [PR-1682](https://github.com/apache/incubator-nuttx/pull/1682) configure.sh
+ can now list configurations with "-L" option
+
+ * [PR-2023](https://github.com/apache/incubator-nuttx/pull/2023) tools: Remove
+ WSL configruation. This is just Linux now.
```
Comment (acassis) [inline ReleaseNotes:27437]: configuration

File: ReleaseNotes [hunk 3]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ SPI transmision (DMA, IOMUX, software CS)
+
+ * [PR-1630](https://github.com/apache/incubator-nuttx/pull/1630) ESP32: Add
+ support for HW RNG
+
+ * [PR-1830](https://github.com/apache/incubator-nuttx/pull/1830) ESP32: Add
+ Power Management of Force-Sleep
+
+ * [PR-1859](https://github.com/apache/incubator-nuttx/pull/1859) ESP32: Add
+ hist timer and improve the oneshot timer logic
```
Comment (acassis) [inline ReleaseNotes:27516]: host

BTW, this patch is for simulator, not for ESP32, it is at wrong section

File: ReleaseNotes [hunk 4]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ Lowputc: Ensure USART is disabled before attempting to configuring it
+
+ * [PR-1714](https://github.com/apache/incubator-nuttx/pull/1714) STM32H7: Fix
+ I2C driver interrupt storm
+
+ * [PR-1556](https://github.com/apache/incubator-nuttx/pull/1556) STM32: Fix IO
+ compentation support in STM32F7 and remove incorrect reference in STM32F0/L0/G0
+
+ * [PR-1529](https://github.com/apache/incubator-nuttx/pull/1529) STM32: Fix
+ intialization bug in ADC that prevented adc\_reset() from working correctly
```
Comment (acassis) [inline ReleaseNotes:27791]: initialization

File: ReleaseNotes [hunk 5]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1561](https://github.com/apache/incubator-nuttx/pull/1561) STM32: Make
+ sure that core over-drive is enable for all chips that support it and operating
```
Comment (acassis) [inline ReleaseNotes:27794]: is enabled

File: ReleaseNotes [hunk 6]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ at 180MHz. Some were enabled at 180MHz but may have not been stable without
+ over-drive not configured.
+
+ * [PR-1553](https://github.com/apache/incubator-nuttx/pull/1553) STM32F7: Fix
+ possible interrupt blocking in serial TXDMA ISR
+
+ * [PR-1544](https://github.com/apache/incubator-nuttx/pull/1544) STM32: Make
+ sure IO compenstation cell is configured prior to call to
```
Comment (acassis) [inline ReleaseNotes:27802]: compensation

File: ReleaseNotes [hunk 7]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ rcc\_enableperipherals() causing syscfg is accessed before it is enabled
+
+ * [PR-1380](https://github.com/apache/incubator-nuttx/pull/1380) STM32F7: Fix
+ tickless driverw where th compare register could be set to a value that has
```
Comment (acassis) [inline ReleaseNotes:27806]: driver

File: ReleaseNotes [hunk 8]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ just passed preventing expiration
+
+ * [PR-1252](https://github.com/apache/incubator-nuttx/pull/1252) STM32L4: Fix
+ 48MHz MSI clock seclection that could cause boot to hang
```
Comment (acassis) [inline ReleaseNotes:27810]: selection

File: ReleaseNotes [hunk 9]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1310](https://github.com/apache/incubator-nuttx/pull/1310) STM32L4:
+ Configure flash wait states earier to prevent corruption of execution state
```
Comment (acassis) [inline ReleaseNotes:27813]: earlier

File: ReleaseNotes [hunk 10]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1248](https://github.com/apache/incubator-nuttx/pull/1248) STM32L4: Fix
+ oneshot timer so that a minimum period is set otherwise it will never be
+ triggered.
+
+ * [PR-1247](https://github.com/apache/incubator-nuttx/pull/1247) STM32L47x/8x:
+ Set additional registers require to place a pin in analog mode
```
Comment (acassis) [inline ReleaseNotes:27820]: required

File: ReleaseNotes [hunk 11]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ of empty pipe with no writers, return EOF instead of EAGAIN
+
+ * [PR-1836](https://github.com/apache/incubator-nuttx/pull/1836) stmpe811: Fix
+ incorrect GPIO interrupt register logic
+
+ * [PR-1741](https://github.com/apache/incubator-nuttx/pull/1741) mmcsd\_sdio:
+ Properly arm the write completion detection
+
+ * [PR-1370](https://github.com/apache/incubator-nuttx/pull/1370) can: Fix
+ incorret usage of nxsem\_getvalue which caused fifo->rx\_sem to increase with
```
Comment (acassis) [inline ReleaseNotes:27851]: incorrect

File: ReleaseNotes [hunk 12]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ SPI support for touch screen controller
+
+ * [PR-1800](https://github.com/apache/incubator-nuttx/pull/1800) vfs: Add
+ `FIOCLEX/FIONCLEX` ioctl support
+
+ * [PR-1798](https://github.com/apache/incubator-nuttx/pull/1798) mmcsd: Allow
+ setting `IDMODE_CLOCK` via kconfig
+
+ * [PR-1587](https://github.com/apache/incubator-nuttx/pull/1587) BCH: Delay the
+ sector flush to avoid multiple earse/write operations in sequence write
```
Comment (acassis) [inline ReleaseNotes:27888]: erase/write

File: ReleaseNotes [hunk 13]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1256](https://github.com/apache/incubator-nuttx/pull/1256) note: Add
+ hooks for note driver for interrupt instrumentation
+
+Board Support -------------
+
+### **Significant Improvements**
+
+ * [PR-1618](https://github.com/apache/incubator-nuttx/pull/1618) metro-m4: Add
+ support for: SmartFS initialization, AT24 EEPROM, GPIO dev, BQ27426 gague
```
Comment (acassis) [inline ReleaseNotes:27932]: gauge

File: ReleaseNotes [hunk 14]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+ where opendir would remove the trailing whitespace or /
+
+ * [PR-1793](https://github.com/apache/incubator-nuttx/pull/1793) vfs: Make sure
+ that rename of mount point uses pseudorename. Previously mv on a mountpoint
+ would return err 88.
+
+ * [PR-1737](https://github.com/apache/incubator-nuttx/pull/1737) vfs: reuse
+ file\_dup2 direction in file\_dup to prevent segfault issue
+
+ * [PR-1490](https://github.com/apache/incubator-nuttx/pull/1490) Insure that
```
Comment (acassis) [inline ReleaseNotes:28002]: Ensure

File: ReleaseNotes [hunk 15]
```diff
@@ -27122,3 +27122,1268 @@ NuttX-9.1.0 Release Notes -------------------------
...
+
+ * [PR-1323](https://github.com/apache/incubator-nuttx/pull/1323) vfs: Add
+ chmod/fchmod/utimes function prototypes
+
+Networking ----------
+
+### **Bug Fixes**
+
+ * [PR-1267](https://github.com/apache/incubator-nuttx/pull/1267) SocketCAN:
+ When timestamped frame was recived while in CAN2.0B mode the frame got dropped.
```
Comment (acassis) [inline ReleaseNotes:28044]: received

**Comment by btashton** [comment]
Thanks @acassis I made all your corrections.  Could someone go ahead and merged this @xiaoxiang781216 @Ouss4.  It is only a text file change so I don't think we need to wait for CI.  I'll also open the PR for the 10.0 branch

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**